Tests/MatterControl.Tests/MatterControl/LibraryProviderTests.cs

```diff
@@ -74,64 +74,64 @@ public void LibraryProviderFileSystem_NavigationWorking()
 			Directory.CreateDirectory(testLibraryDirectory);
 
 			LibraryProviderFileSystem testProvider = new LibraryProviderFileSystem(testLibraryDirectory, "TestPath", null, null);
-			testProvider.DataReloaded += (s, e) => { dataReloaded = true; };
+			testProvider.DataReloaded += (s, e) => { dataReloadedCount++; };
+
+			MatterControlUtilities.SleepUntil(() => { return dataReloadedCount > 0; }, 1);
+			dataReloadedCount = 0;
 
 			Assert.IsTrue(testProvider.CollectionCount == 0, "Start with a new database for these tests.");
 			Assert.IsTrue(testProvider.ItemCount == 0, "Start with a new database for these tests.");
 
 			// create a collection and make sure it is on disk
-			dataReloaded = false; // it has been loaded for the default set of parts
+			Assert.AreEqual(0, dataReloadedCount); // it has been loaded for the default set of parts
 
 			string collectionName = "Collection1";
 			string createdDirectory = Path.Combine(testLibraryDirectory, collectionName);
 
 			Assert.IsFalse(Directory.Exists(createdDirectory), "CreatedDirectory should *not* exist");
-			Assert.IsFalse(dataReloaded, "Reload should *not* have occurred");
+			Assert.AreEqual(0, dataReloadedCount, "Reload should *not* have occurred");
 
 			testProvider.AddCollectionToLibrary(collectionName);
-			Thread.Sleep(500); // wait for the add to finish
+			MatterControlUtilities.SleepUntil(() => { return testProvider.CollectionCount == 1; }, 1);
```

Comment:
These SleepUntil lamdas might look cleaner with statements rather than blocks:

MatterControlUtilities.SleepUntil(() => testProvider.CollectionCount == 1, 1);
versus
MatterControlUtilities.SleepUntil(() => { return testProvider.CollectionCount == 1; }, 1);